### DISCUSSION TITLE
Allow custom cli parameters for 'remix.init'

### DISCUSSION THREAD

**Discussion by KnisterPeter** [remix-run/remix#7181] [Proposals]
We are building a stack for our company internal apps. This stack has some options devs need to answer during the stack initialization. So far so good, we use inquirer for that.

But for the sake of automating this (e.g. to use it in e2e tests for the stack) we would like to have an option to add custom cli parameters to the `create-remix` command. Like `--install` or `--typescript` these could be used instead of asking with inquirer.

Currently the create command fails with the error `unknown or unexpected option` and does not execute at all.

**Comment by kiliman** [comment]
In the meantime, a workaround is to use environment variables.

```bash
REMIX_INIT_ARG="Hello World" npx create-remix --template PATH
```

```js
// remix.init/index.js
async function main({ isTypeScript, packageManager, rootDirectory }) {
  const arg = process.env.REMIX_INIT_ARG ?? "some default";
  console.log("remix.init", arg);
}

module.exports = main;
```
